pytorch_lightning/strategies/ddp.py

```diff
@@ -137,8 +137,8 @@ def process_group_backend(self) -> Optional[str]:
         return self._process_group_backend
 
     def _configure_launcher(self) -> None:
-        self._launcher = _SubprocessScriptLauncher(self.cluster_environment, self.num_processes, self.num_nodes)
         if not self.cluster_environment.creates_processes_externally:
+            self._launcher = _SubprocessScriptLauncher(self.cluster_environment, self.num_processes, self.num_nodes)
             self._rank_0_will_call_children_scripts = True
```

Comment:
noob question unrelated to the PR:
why is this attribute not part of the launcher?

Reply:
Because the value of this variable has to be broadcast by the strategy and the deadlock detection is currently part of the strategy. At the time of extracting the launcher out of the strategy, there was too much complexity to separate this cleanly. This has to be revisited if we want to combine ddp and ddp-spawn into one strategy.